h2o-automl/src/main/java/ai/h2o/automl/targetencoding/TargetEncoderModel.java

```diff
@@ -0,0 +1,158 @@
+package ai.h2o.automl.targetencoding;
+
+import hex.Model;
+import hex.ModelCategory;
+import hex.ModelMetrics;
+import hex.ModelMojoWriter;
+import water.H2O;
+import water.Iced;
+import water.Key;
+import water.MRTask;
+import water.fvec.Chunk;
+import water.fvec.Frame;
+import water.util.IcedHashMap;
+
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class TargetEncoderModel extends Model<TargetEncoderModel, TargetEncoderModel.TargetEncoderParameters, TargetEncoderModel.TargetEncoderOutput> {
+
+  public TargetEncoderModel(Key<TargetEncoderModel> selfKey, TargetEncoderParameters parms, TargetEncoderOutput output) {
+    super(selfKey, parms, output);
+  }
+
+  @Override
+  public ModelMetrics.MetricBuilder makeMetricBuilder(String[] domain) {
+    throw H2O.unimpl("No Model Metrics for TargetEncoder.");
+  }
+
+  @Override
+  protected double[] score0(double[] data, double[] preds) {
+    return new double[0];
+  }
+
+  private static IcedHashMap<String, Map<String, TEComponents>> _targetEncodingMap;
```

Comment:
Why is it `static`? We can't work with multiple target encoders at the same time?

Reply:
Ouch! Good catch @mn-mikke!

Should be fixed.

Reply:
@michalkurka  I was trying to make integration for SW faster. I believe for that goal only API is needed. I quickly prepared a version for @mn-mikke  that has API and which could be wrapped in SW.  As a result we just wasting time commenting "in progress" code. I think giving a link to a PR would be enough next time. 
Static code was used to make it faster as some encoding map conversion logic was still in other places and was supposed to be moved to a more appropriate ones.

Reply:
I see this is still WIP. Got it - in that case, things can still change. Np.